src/SQLStore/SimpleListLookup.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace SMW\SQLStore;
+
+/**
+ * A basic interface for fetching a simple list either from a DB or cache
+ * instance
+ *
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+interface SimpleListLookup {
```

Comment:
Surprise: I find this interface a bit strange. (As I always do right? :))

It has a single method inherently relevant to list lookup stuff. The other 3 make sense in caching context. I do not think they belong to the base interface. For what reasons did you put them there?

One reason could be that `ListLookupCache` is a decorator, and that you want other caching implementations of the base interface to work correctly inside there. I'm not sure why that'd be needed though, esp since an alternate cache implementation would be odd due to the current usage of a caching strategy. 

How about dropping the caching related methods from `SimpleListLookup`, and moving that responsibility into `ListLookupCache`? This thing itself can get the timestamp, and it can create or require (in the constructor) an unique identifier.